modin/core/execution/unidist/implementations/pandas_on_unidist/partitioning/partition_manager.py

```diff
@@ -0,0 +1,172 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Module houses class that implements ``GenericUnidistDataframePartitionManager`` using Unidist."""
+
+import inspect
+import threading
+
+import unidist
+
+from modin.config import ProgressBar
+from modin.core.execution.unidist.generic.partitioning import (
+    GenericUnidistDataframePartitionManager,
+)
+from modin.core.execution.unidist.common import UnidistWrapper
+from .virtual_partition import (
+    PandasOnUnidistDataframeColumnPartition,
+    PandasOnUnidistDataframeRowPartition,
+)
+from .partition import PandasOnUnidistDataframePartition
+from modin.core.execution.unidist.generic.modin_aqp import call_progress_bar
+
+
+def progress_bar_wrapper(f):
```

Comment:
can we just import it from ray's partition manager? the functions look identical

Reply:
I do not want unidist to depend on other Modin engines that may require their installation. So I prefer to leave as is.

Reply:
https://github.com/modin-project/modin/issues/5275